diff --git a/main.py b/main.py
index e738df3..c447a68 100644
--- a/main.py
+++ b/main.py
@@ -39,7 +39,7 @@ def parse_args():
     parser.add_argument("--torch-deterministic", type= lambda x:bool(strtobool(x)), default=True, nargs="?", const=True, help="if toggeled, 'torch-backends.cudnn.deterministic=False'")
     parser.add_argument("--cuda", type=lambda x: bool(strtobool(x)), default=True, nargs="?", const=True, help="if toggled, cuda will be enabled by default")
     parser.add_argument("--wandb-track", type=lambda x: bool(strtobool(x)), default=False, help="if toggled, this experiment will be tracked with Weights and Biases project")
-    parser.add_argument("--wandb-name", type=str, default="UAV_Subband_Allocation_DQN_Pytorch", help="project name in Weight and Biases")
+    parser.add_argument("--wandb-name", type=str, default="uav_nw_correlated_dql_connectivity", help="project name in Weight and Biases")
     parser.add_argument("--wandb-entity", type=str, default= None, help="entity(team) for Weights and Biases project")
 
     # Arguments specific to the Algotithm used 
@@ -172,7 +172,7 @@ class DQL:
         # Joint action size = number of agents ^ action size // for a state 
         # Optimizing the joint action so setting as a variable for CE optimization 
         joint_action_size = NUM_UAV ** UAV_OB[agent_idx].action_size
-        prob_weight = Variable(joint_action_size, pos = True)
+        prob_weight = Variable(joint_action_size, boolean = True)
         
         # Collect Q values for the corresponding states of each individual agents
         # Using negate value to use Minimize function for solving  // removed
@@ -210,7 +210,7 @@ class DQL:
                 weights = prob_weight.value
                 # print(weights)
                 # print('Max Weight:', np.max(weights))
-                # print("Best Joint Action:", np.argmax(weights))
+                print("Best Joint Action:", np.argmax(weights))
             else:
                 weights = None
         except:
diff --git a/run.py b/run.py
index 5ccae10..722c7f0 100644
--- a/run.py
+++ b/run.py
@@ -3,7 +3,9 @@ import subprocess
 #Run the script for multiple levels of information exchange from level 1 to level 4
 for i in range(1, 5):
     print('#######################################################')
-    print('####  Running the code for Level:', i, "info exchange  ####")
+    print('####              Running the code                 ####')
     print('#######################################################')
     p = subprocess.run(["python", "uav_env.py"])
-    g = subprocess.run(["python", "main.py", "--info-exchange-lvl", str(i), "--num-episode", str(600), "--wandb-track", "True", "--learning-rate", str(2.5e-4)])
+    g = subprocess.run(["python", "main.py", "--num-episode", str(50), "--wandb-track", "True", "--learning-rate", str(2.5e-4)])
+
+# python main.py --num-episode 50 --wandb-track True --learning-rate 2.5e-4
\ No newline at end of file
